src/magic_folder/downloader.py

```diff
@@ -489,18 +529,29 @@ class InMemoryMagicFolderFilesystem(object):
 
     def __init__(self):
         self.actions = []
+        self._staged_content = {}
```

Comment:
What is this for? Nothing ever deletes entries in it

Reply:
This is to ensure that the object passed to `mark_overwrite` matches the object returned from the corresponding `download_content_to_staging`. It is a mapping from `id(remote_snapshot)` (since `RemoteSnapshot`'s are not hashable) to marker objects.

Reply:
Nothing ever deletes from it though.

Reply:
I'm not sure why you think it is obvious that something should. After thinking about if for a while, I guess it would make sense to remove the entry in `mark_overwrite`/`mark_conflict`, to allow verifying that we don't pass the same file twice. That would catch errors that the current code doesn't. However, as a test-double , the lack of that doesn't break anything, and it is checking more than previously.

I think I originally added this because I at one point, I had `download_content_to_staging` returning a tuple, and wanted to ensure that the right bit of it got passed to `mark_overwrite`, other ways of misusing it were not on my mind. 

In any case, I think the shape of this abstraction may need to be changed for #454 (I think the code that touches the filesystem needs to be more involved with deciding whether there is a conflict), so I think we can live with not catching the additional error in the double.